tests/should_succeed/ite_normalize.rs

```diff
@@ -76,24 +76,43 @@ extern_spec! {
     }
 }
 
-impl From<usize> for Expr {
+impl ::std::convert::From<usize> for Expr {
+    #[ensures(result.comes_from(a))]
     fn from(a: usize) -> Self {
         Self::variable(a)
     }
 }
 
-impl From<bool> for Expr {
+impl ::std::convert::From<bool> for Expr {
+    #[ensures(result.comes_from(b))]
     fn from(b: bool) -> Self {
         if b { Self::True } else { Self::False }
     }
 }
 
+impl From<usize> for Expr {
+    #[predicate]
+    #[open]
+    fn comes_from(self, v: usize) -> bool {
+        self == Expr::Var { v }
```

Comment:
Alternatively you could use `true` here. That would be closer to the current code.